sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -153,7 +154,7 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
             // If we have the same paths, same schema, and same partition spec,
             // we will use the cached relation.
             val useCached =
-              relation.location.rootPaths.toSet == Set(metastoreRelationRootPath) &&
```

Comment:
It looks like we can get rid of the local `val metastoreRelationRootPath = metastoreRelation.hiveQlTable.getDataLocation` above.